depthai_helpers/managers.py

```diff
@@ -37,52 +115,130 @@ class Previews(enum.Enum):
     right = partial(lambda packet, _: packet.getCvFrame())
     rectified_left = partial(lambda packet, _: cv2.flip(packet.getCvFrame(), 1))
     rectified_right = partial(lambda packet, _: cv2.flip(packet.getCvFrame(), 1))
-    depth = partial(convert_depth_frame)
+    depth_raw = partial(lambda packet, _: packet.getFrame())
+    depth = partial(convert_depth_raw_to_depth)
     disparity = partial(convert_disparity_frame)
     disparity_color = partial(convert_disparity_to_color)
 
 
+class MouseClickTracker:
+    def __init__(self):
+        self.points = {}
+        self.values = {}
+
+    def select_point(self, name):
+        def cb(event, x, y, flags, param):
+            if event == cv2.EVENT_LBUTTONUP:
+                self.values = {}
+                if self.points.get(name) == (x, y):
+                    del self.points[name]
+                else:
+                    self.points[name] = (x, y)
+        return cb
+
+    def extract_value(self, name, frame: np.ndarray):
+        point = self.points.get(name, None)
+        if point is not None:
+            try:
+                if name in (Previews.depth_raw.name, Previews.depth.name):
+                    self.values[name] = "{}mm".format(frame[point[1]][point[0]])
+                elif name in (Previews.disparity_color.name, Previews.disparity.name):
+                    self.values[name] = "{}px".format(frame[point[1]][point[0]])
+                elif frame.shape[2] == 3:
+                    self.values[name] = "R:{},G:{},B:{}".format(*frame[point[1]][point[0]][::-1])
+                else:
+                    self.values[name] = str(frame[point[1]][point[0]])
+            except:
+                pass
+
+
 class PreviewManager:
-    def __init__(self, fps, display, colorMap=cv2.COLORMAP_JET, disp_multiplier=255/96):
+    def __init__(self, fps, display, colorMap=cv2.COLORMAP_JET, disp_multiplier=255/96, mouseTracker=False):
         self.display = display
         self.frames = {}
         self.raw_frames = {}
         self.fps = fps
         self.colorMap = colorMap
         self.disp_multiplier = disp_multiplier
-
-    def create_queues(self, device):
-        def get_output_queue(name):
+        self.mouse_tracker = MouseClickTracker() if mouseTracker else None
+
+    def create_queues(self, device, callback=lambda *a, **k: None):
+        calib = device.readCalibration()
+        eeprom = calib.getEepromData()
+        cam_info = eeprom.cameraData[calib.getStereoLeftCameraId()]
```

Comment:
This will fail on OAK1, or if calibration is empty/corrupted.

Reply:
From previous review comments:
```
Baseline and focal length should be obtained from EEPROM data, if available, otherwise defaulted to these values.
This is not going to work well on smaller baseline boards.
```

Reply:
Ok, updated so that OAK-1 works without issues with this script too.
> Baseline and focal length should be obtained from EEPROM data, if available, otherwise defaulted to these values.
> This is not going to work well on smaller baseline boards.
when the calibration is empty,  `readCalibration` or `getEepromData` would throw an error or return null @szabi-luxonis?